lua/entities/gmod_tardis_interior/modules/sh_interior_lights.lua

```diff
@@ -0,0 +1,150 @@
+cvars.AddChangeCallback("tardis2_debug_lamps", function()
+    TARDIS.debug_lamps = GetConVar("tardis2_debug_lamps"):GetBool()
+end)
+
+TARDIS.debug_lamps = GetConVar("tardis2_debug_lamps"):GetBool()
+
+if CLIENT then
+
+    function ENT:DrawLight(id,light)
+        if self:CallHook("ShouldDrawLight",id,light)==false then return end
+        local dlight = DynamicLight(id, true)
+        if ( dlight ) then
+            local size=1024
+            local c=light.color
+            local warnc = light.warncolor or c
+            local warning = self.exterior:GetData("health-warning", false)
+            dlight.Pos = self:LocalToWorld(light.pos)
+            dlight.r = warning and warnc.r or c.r
+            dlight.g = warning and warnc.g or c.g
+            dlight.b = warning and warnc.b or c.b
+            dlight.Brightness = light.brightness
+            dlight.Decay = size * 5
+            dlight.Size = size
+            dlight.DieTime = CurTime() + 1
+        end
+    end
+
+    ENT:AddHook("Think", "lights", function(self)
+        if TARDIS:GetSetting("lightoverride-enabled") then return end
+        local light=self.metadata.Interior.Light
+        local lights=self.metadata.Interior.Lights
+        local index=self:EntIndex()
+        if light then
+            self:DrawLight(index,light)
+        end
+        if lights then
+            local i=0
+            for _,light in pairs(lights) do
+                i=i+1
+                self:DrawLight((index*1000)+i,light)
+            end
+        end
+    end)
+
+    function ENT:AddRoundThing(pos)
+        self.roundthings[pos]=util.GetPixelVisibleHandle()
+    end
+
+    ENT:AddHook("Initialize", "lights-roundthings", function(self)
+        if self.metadata.Interior.RoundThings then
+            self.roundthingmat=Material("sprites/light_ignorez")
+            self.roundthings={}
+            for k,v in pairs(self.metadata.Interior.RoundThings) do
+                self:AddRoundThing(v)
+            end
+        end
+    end)
+
+    local size=32
+    ENT:AddHook("Draw", "lights-roundthings", function(self)
+        if self.roundthings then
+            if self:CallHook("ShouldDrawLight")==false then return end
+            for k,v in pairs(self.roundthings) do
+                local pos = self:LocalToWorld(k)
+                local vis = util.PixelVisible(pos, 3, v)*255
+                if vis > 0 then
+                    render.SetMaterial(self.roundthingmat)
+                    render.DrawSprite(pos, size, size, Color(255,153,0, vis))
+                end
+            end
+        end
+    end)
+
+else -- SERVER
+    
+    if SERVER then
+        ENT:AddHook("Initialize", "lamps", function(self)
+            local lamps = self.metadata.Interior.Lamps
+            if not lamps then return end
+
+            self.lamps = {}
+
+            for k,v in pairs(lamps) do
+                if v then
+                    local lamp = MakeLamp(nil, --self:GetCreator(),
```

Comment:
Why is self:GetCreator() commented out here?

Reply:
that's the "owner" parameter for the entity
I got rid of that due to the limit of entities per person
(if all TARDIS parts were owned by a player, he wouldn't be able to spawn many)

Reply:
Will close this then as it's gonna need restructuring for the clientside stuff as we talked about